### ISSUE TITLE
System crash when rescanning the titles

### ISSUE THREAD

**Issue by muxi1** [J-D-K/JKSV#32]
Hello @J-D-K 
The Title Rescan option in version 05.20.2020 causes a system crash. 

![err](https://user-images.githubusercontent.com/39792666/82669146-2ec6b780-9c3b-11ea-98f8-f152711b7699.jpg)

**Comment by J-D-K** [comment]
Is it consistent, or are you enabling options first? There are two that would make it load more than before.

**Comment by muxi1** [comment]
I don't understand exactly what you mean by your question, but as soon as I start JKSV and select Title Rescan in the options (ZR), the system crashes.

**Comment by J-D-K** [comment]
That actually answers it for me. I meant that in the options menu, there's two options. Force Mount and Account Sys Saves. Both of which will change what is ignored while loading.

Is there a reason you're rescanning? It's only really there for those options and when you terminate something to see what else you can get into. I'll get back to ya with a build that'll tell me something when I can.

**Comment by muxi1** [comment]
No, there was no need for me to rescan. I just wanted to test the function and that's what caused the system to crash.

**Comment by J-D-K** [comment]
No, it shouldn't cause any crashes. That's the way JKSV has always ran. It ignored things that couldn't be opened and couldn't open the account system stuff cause of my own mistake. I've tweaked something in this build and added code to sort of log the rescan function. It should give me a rough idea of where it's getting caught. Upload the `JKSV/log.txt` to pastebin or something. Logging JKSV's title loading can get massive, so I'm only catching title ids and what user is returned.
[JKSV.zip](https://github.com/J-D-K/JKSV/files/4670705/JKSV.zip)

**Comment by muxi1** [comment]
Hi,
here is the log file. I hope it's helpful.

[log.txt](https://github.com/J-D-K/JKSV/files/4671743/log.txt)

**Comment by J-D-K** [comment]
Thanks. It's what I suspected. It's finding a new user in rescan which is very odd.

**Comment by muxi1** [comment]
I have only set up a single user.

**Comment by J-D-K** [comment]
It's finding something new. I don't know what yet. -1 is returned when a user hasn't been loaded before. It's probably system stuff.

**Comment by muxi1** [comment]
I've previously restored a save from ACNH from another console, but that certainly wasn't the reason, was it?

**Comment by J-D-K** [comment]
I don't believe so. This build should get around it, but there is something the first one is skipping that the stripped down version sees. I wonder what it is?
[JKSV.zip](https://github.com/J-D-K/JKSV/files/4671937/JKSV.zip)

**Comment by muxi1** [comment]
With this build the system no longer crashes. Did you just disable this feature or fix the problem?

**Comment by J-D-K** [comment]
I didn't disable the rescan. I added code to bypass the problem until I have time to properly fix it.

**Comment by muxi1** [comment]
Ah ok, that makes sense.

**Comment by J-D-K** [comment]
Let me know how this goes. I cleaned up the data file a lot. The initial load and rescan use the same function. I'm more curious about if this one will catch the problem user thing.
[JKSV - 05.24.2020.zip](https://github.com/J-D-K/JKSV/files/4674635/JKSV.-.05.24.2020.zip)

**Comment by muxi1** [comment]
I can't judge if the title rescan function works, but even with this new version there is at least no system crash when it is executed. What's the point of this option shown here? This is probably new, since it was not available in the last version.

![jksv](https://user-images.githubusercontent.com/39792666/82787282-5f3e6980-9e66-11ea-99d4-e65bf2937d93.jpg)

**Comment by J-D-K** [comment]
That would be the mystery user account that was making it crash before. Is there anything in it? Or, do you get the nothing found pop up?

**Comment by muxi1** [comment]
DLCs from Doom 1993 are stored there.

**Comment by J-D-K** [comment]
Neat. Does the folder menu tell you what save data type it is? I actually added code for all of them, but I only have code to make fake users for device saves and the other two.

**Comment by J-D-K** [comment]
Also, congrats. We found something new JKSV can do. Yay

**Comment by muxi1** [comment]
Here is the content of one of the 6 icons. The data are all identical except for the SiD. The DLCs of Doom 1993 are probably stored differently, as it is usually done.

![jksv](https://user-images.githubusercontent.com/39792666/82789781-31a7ef00-9e6b-11ea-82d7-701874dff083.jpg)

**Comment by J-D-K** [comment]
I've added code to handle the rest of the save data types and add a fake user _if_ they exist. Well, I think I did. I don't have any to check. If this picks it up and registers it as a user named "Cache", I'm going to consider this fixed. If it isn't crashing on the original title load, the rescan won't either since they're using the same code now.
[JKSV - 05.25.2020.zip](https://github.com/J-D-K/JKSV/files/4678454/JKSV.-.05.25.2020.zip)
